openedx/core/djangoapps/content/learning_sequences/api/outlines.py

```diff
@@ -97,31 +98,35 @@ def get_course_outline(course_key: CourseKey) -> CourseOutlineData:
     ]
 
     outline_data = CourseOutlineData(
-        course_key=learning_context.context_key,
-        title=learning_context.title,
-        published_at=learning_context.published_at,
-        published_version=learning_context.published_version,
+        course_key=course_context.learning_context.context_key,
+        title=course_context.learning_context.title,
+        published_at=course_context.learning_context.published_at,
+        published_version=course_context.learning_context.published_version,
         sections=sections_data,
+        course_visibility=course_context.course_visibility,
     )
     TieredCache.set_all_tiers(cache_key, outline_data, 300)
 
     return outline_data
 
 
-def _get_learning_context_for_outline(course_key: CourseKey) -> LearningContext:
+def _get_course_context_for_outline(course_key: CourseKey) -> CourseContext:
+    """
+    Get Course Context for given param:course_key
+    """
     if course_key.deprecated:
         raise ValueError(
             "Learning Sequence API does not support Old Mongo courses: {}"
             .format(course_key),
         )
     try:
-        learning_context = LearningContext.objects.get(context_key=course_key)
+        course_context = LearningContext.objects.get(context_key=course_key).course_context
```

Comment:
Nit: I think you can bring this from being two queries back down to one by doing the get on the `CourseContext` and doing a `select_related` to fetch its `learning_context` at the same time.

Reply:
Wow. I am learning a lot. Thanks @ormsbee this is unknown to me. It really get queries down to one :+1: